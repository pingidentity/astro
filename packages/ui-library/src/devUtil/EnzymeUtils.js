import { mount } from "enzyme";


export const snapshotDataIds = node => {
    const elements = node.find("[data-id]").hostNodes(); // Find only actual HTML elements

    const ids = elements.map(element => element.prop("data-id")).filter(item => item !== "");
    expect(ids).toMatchSnapshot();
};

export const mountSnapshotDataIds = node => snapshotDataIds(mount(node));
