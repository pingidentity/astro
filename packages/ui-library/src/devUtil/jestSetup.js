
// This is needed for testing the modal and DetailsTooltip component.
global.getSelection = jest.fn();
global.getSelection.mockReturnValue({
    toString: () => "",
});
global.document.createRange = () => ({
    setStart: () => {},
    setEnd: () => {},
    commonAncestorContainer: {
        nodeName: "BODY",
        ownerDocument: document,
    },
});
