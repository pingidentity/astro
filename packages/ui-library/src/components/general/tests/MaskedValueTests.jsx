jest.dontMock("../Link");

import { mountSnapshotDataIds } from "../../../devUtil/EnzymeUtils";

describe("Masked Value", function() {
    var React = require("react"),
        ReactTestUtils = require("react-dom/test-utils"),
        TestUtils = require("../../../testutil/TestUtils"),
        MaskedValue = require("../MaskedValue");

    it("data-id's don't change", () => {
        mountSnapshotDataIds(
            <MaskedValue />
        );
    });

    it("renders with a string and toggles", function() {
        var component = ReactTestUtils.renderIntoDocument(<MaskedValue>A value</MaskedValue>);

        const toggler = TestUtils.findRenderedDOMNodeWithDataId(component, "reveal");
        ReactTestUtils.Simulate.click(toggler);
        expect(component.state.maskValue).toBe(false);

        expect(ReactTestUtils.isDOMComponent(component)).toBeDefined();
    });

    it("renders with a node", function() {
        var component = ReactTestUtils.renderIntoDocument(
            <MaskedValue>
                <p>yes</p>
            </MaskedValue>
        );
        expect(ReactTestUtils.isDOMComponent(component)).toBeDefined();
    });

    it("renders with multiple nodes", function() {
        var component = ReactTestUtils.renderIntoDocument(
            <MaskedValue>
                <p>yes</p>
                <p>no</p>
            </MaskedValue>
        );
        expect(ReactTestUtils.isDOMComponent(component)).toBeDefined();
    });

    it("triggers the right toggle callback", function() {
        const callback = jest.fn();
        const component = ReactTestUtils.renderIntoDocument(
            <div>
                <MaskedValue onToggleReveal={callback} maskValue={true} />
            </div>
        );

        const toggler = TestUtils.findRenderedDOMNodeWithDataId(component, "reveal");

        expect(callback).not.toBeCalled();
        ReactTestUtils.Simulate.click(toggler);
        expect(callback).toBeCalled();
    });
});
