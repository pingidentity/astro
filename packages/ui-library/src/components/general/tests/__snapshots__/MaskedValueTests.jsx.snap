// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Masked Value data-id's don't change 1`] = `
Array [
  "masked-value",
  "reveal",
]
`;
